watertap/unit_models/crystallizer.py

```diff
@@ -0,0 +1,696 @@
+        default=False,
+        domain=In([False]),
+        description="Holdup construction flag - must be False",
+        doc="""Indicates whether holdup terms should be constructed or not.
+    **default** - False. The filtration unit does not have defined volume, thus
+    this must be False."""))
+
+    CONFIG.declare("property_package", ConfigValue(
+        default=useDefault,
+        domain=is_physical_parameter_block,
+        description="Property package to use for control volume",
+        doc="""Property parameter object used to define property calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}"""))
+
+    CONFIG.declare("property_package_args", ConfigBlock(
+        implicit=True,
+        description="Arguments to use for constructing property packages",
+        doc="""A ConfigBlock with arguments to be passed to a property block(s)
+    and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}"""))
+
+
+    def build(self):
+        super().build()
+
+        solvent_set = self.config.property_package.solvent_set
+        solute_set = self.config.property_package.solute_set
+
+        # this creates blank scaling factors, which are populated later
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        # Next, get the base units of measurement from the property definition
+        units_meta = self.config.property_package.get_metadata().get_derived_units
+
+        # Add unit variables
+
+        self.hex_max_temp_increase = Param(
+            initialize=4,
+            units=pyunits.K,
+            doc="Maximum temperature difference between inlet and outlet of a crystallizer heat exchanger.\
+            Lewis et al. suggests 1-2 degC but use 5degC in example; Tavare example used 4 degC.\
+            Default is 4 degC")
+
+        # ====== Crystallizer sizing parameters ================= #
+        self.L_d = Param(
+            initialize=0.5e-3, # From Mersmann et al., Tavare et al. example
+            # bounds=(0.2e-3, 0.6e-3), # Limits for FC crystallizers based on Bermingham et al.
+            units=pyunits.m,
+            doc="Desired median crystal size, m"
+            )
+
+        self.G_rate = Param(
+            initialize=3.7e-8, # From Mersmann et al. for NaCl. Perry has values between 0.5e-8 to 13e-8 for NaCl
+            # bounds=(1e-9, 1e-6), # Based on Mersmann and Kind diagram. 
+            units=pyunits.m/pyunits.s,
+            doc="Crystal growth rate, m/s"
+            )
+
+
+        self.k_param = Param(
+            initialize=0.04,
+            units=pyunits.m/pyunits.s,
+            doc="Sounders-Brown constant, set at 0.04 m/s based on Dutta et al. \
+            Lewis et al suggests 0.024 m/s, while Tavare suggests about 0.06 m/s ")
```

Comment:
We _could_ set these as variables and then fix them here in the unit model (similar to what we do in property models). That way, if we wanted to use parameter estimation later on, it wouldn't be an issue.

Reply:
Done